File: 2/1.0/C#/Program.cs
using System.IO;
using System;


namespace lab_1
{
    public class Program
    {
        static void Main(string[] args)
        {
            Console.OutputEncoding = System.Text.Encoding.Unicode;
            Console.InputEncoding = System.Text.Encoding.Unicode;

            string text = InputText();      //Введення тексту від користувача

            WriteToFile(@"D:\VSproj\Lab_1\Files\First.txt", text);  //Створення і запис тексту в текстовий файл

            Console.WriteLine();

            Console.WriteLine("Вміст створеного файлу: ");

            ReadFromFile(@"D:\VSproj\Lab_1\Files\First.txt");       //Відкриття та читання вмісту відповідного текстового файлу

            string newText = ChangeText(GetTextFromFile(@"D:\VSproj\Lab_1\Files\First.txt"));      //Зміна тексту згідно із завдванням

            WriteToFile(@"D:\VSproj\Lab_1\Files\Second.txt", newText);      //Стоврення та запис зміненного тексту у новий файл.

            Console.WriteLine();

            Console.WriteLine("Вміст нового файлу: ");

            ReadFromFile(@"D:\VSproj\Lab_1\Files\Second.txt");           //Відкриття та читання вмісту відповідного текстового файлу
        }

        static void WriteToFile(string path, string text)       //Метод для створення та запису текста у файл
        {
            StreamWriter sw = Check(path);

            foreach (string line in text.Split('\n'))
                sw.WriteLine(line);

            sw.Close();
        }

        static string GetTextFromFile(string path)                  //Метод для зчитування тексту з файла
        {
            string result = String.Empty;

            foreach (string line in File.ReadAllLines(path))
                result += line + '\n';

            result = result.Remove(result.Length - 1);

            return result;
        }

        static void ReadFromFile(string path)                           //Метод для зчитування тексту з файлу
        {
            foreach(string line in File.ReadAllLines(path))
                Console.WriteLine(line);
        }

        static string ChangeText(string text)                   //Метод для зміни тексту
        {
            string result = String.Empty;

            string resultBegin = String.Empty;
            string resultEnd = String.Empty;

            foreach (string line in text.Split('\n'))
            {
                if (line.StartsWith('#'))                           //Зберігання рядків, які починаються з # в одну змінну, решта - в іншу
                    resultEnd += line.Remove(0, 1) + '\n';
                else
                    resultBegin += line + '\n';
            }

            result += resultBegin;

            foreach (string line in resultEnd.Split('\n'))
                result += line.Insert(line.Length / 2, "!") + '\n';         //Додавання знака оклику у середину рядка

            result = result.Remove(result.Length - 2);  

            return result;
        }

        static string InputText()                                             //Метод для введення тексту від користувача
        {
            Console.WriteLine("Введіть текст для подальших дій: ");

            string? text_result = String.Empty;
            string? temp_text = String.Empty;

            while (true)
            {
                temp_text = Console.ReadLine();                         
                    
                if (temp_text.Trim()[0] != 4)                           //Ввод здійснюється доки користувач не введе Ctrl + D
                {
                    text_result += temp_text + '\n';
                    continue;
                }
                else
                    break;   
            }

            text_result = text_result.Remove(text_result.Length - 1);

            return text_result;
        }

        static StreamWriter Check(string path)                           //Якщо текстовий документ вже інсує, то він доповнюється. Якщо не існує, то створюється
        {
            StreamWriter sw;

            if (File.Exists(path))
                sw = new StreamWriter(path, true, System.Text.Encoding.Unicode);
            else
                sw = new StreamWriter(path, false, System.Text.Encoding.Unicode);

            return sw;
        }
    }
}  

File: 2/1.0/Python/Program.py
import os


def input_text():                   #Ввод тексту від користувача
    result = ""
    temp_str = ""

    print("Введіть текст для подальших дій: ")

    while True:

        temp_str = input()          

        if temp_str != chr(4):                  #Закінчення вводу тексту при Ctrl + D
            result += temp_str + '\n'
        else:
            break

    result = result.removesuffix('\n')

    return result


def get_text_from_file(path):                   #Отримання тексту з файлу
    result = ""

    file = open(path, 'r')

    for line in file.readlines():
        result += line

    return result


def write_to_file(path, text):                 #Запис тексту у файл

    if os.path.exists(path):
        file = open(path, 'a')
        file.write('\n')
        file.write(text)
    else:
        file = open(path, 'w')
        file.write(text)

    file.close()


def read_from_file(path):                       #Читання тексту з файлу
    file = open(path, 'r')

    for line in file.readlines():
        print(line.removesuffix('\n'))

    file.close()

        
def change_text(text):                          #Зміна тексту відповідно до умови
    result = ""
    result_begin = ""
    result_end = ""

    for line in text.split('\n'):
        if line.startswith('#'):
            result_end += line.removeprefix('#') + '\n'
        else:
            result_begin += line + '\n'
    
    result_end = result_end.removesuffix('\n')

    result += result_begin

    for line in result_end.split('\n'):
        result += line[:int(len(line) / 2)] + '!' + line[int(len(line) / 2):] + '\n'

    result = result.removesuffix("\n")

    return result


def write_empty_line(path):                         #Перевод запису тексту у файлі на наступний рядок
    file = open(path, 'a')
    file.write('\n')
    file.close()



def main():

    text = input_text()

    write_to_file(r"D:\VSproj\Lab_1\Files\First.txt", text)

    print()

    print("Вміст текстового файлу: ")

    read_from_file(r"D:\VSproj\Lab_1\Files\First.txt")

    text = change_text(get_text_from_file(r"D:\VSproj\Lab_1\Files\First.txt"))

    write_to_file(r"D:\VSproj\Lab_1\Files\Second.txt", text)

    print()

    print("Вміст нового текстового файлу: ")

    read_from_file(r"D:\VSproj\Lab_1\Files\Second.txt")

    write_empty_line(r"D:\VSproj\Lab_1\Files\Second.txt")


if __name__ == "__main__":
    main()